models/iaf_cond_alpha_mc.cpp

```diff
@@ -117,13 +117,39 @@ nest::iaf_cond_alpha_mc_dynamics( double, const double y[], double f[], void* pn
   assert( pnode );
   const nest::iaf_cond_alpha_mc& node = *( reinterpret_cast< nest::iaf_cond_alpha_mc* >( pnode ) );
 
+  bool is_refractory = false;
+
   // compute dynamics for each compartment
   // computations written quite explicitly for clarity, assume compile
   // will optimized most stuff away ...
   for ( size_t n = 0; n < N::NCOMP; ++n )
   {
-    // membrane potential for current compartment
-    const double V = y[ S::idx( n, S::V_M ) ];
+    // membrane potential for current compartment and coupling currents
+    double V;
+    double I_conn;
+    switch ( n )
+    {
+    case N::SOMA:
+    {
+      is_refractory = node.S_.r_ > 0;
+      V = is_refractory ? node.P_.V_reset : std::min( y[ S::idx( N::SOMA, S::V_M ) ], node.P_.V_th );
```

Comment:
The models that do not have the min() also do not have a threshold parameter as far as I could work out, so that is why they are missing the min(). Is there some other parameter that I can change it with, so instead of the threshold we use something else?

Reply:
You're right, we should not change models for which these aren't straightforwardly defined. Could you just look at iaf_chxk_2008? It has refractoriness and a threshold voltage parameter.

Reply:
iaf_chxk_2008 does not have explicit refractoriness or membrane potential reset. When the threshold is crossed, this model activates a conductances that pulls the membrane potential down again (g_AHP). But the membrane potential can actually remain above threshold for some (essentially arbitrary) time. A spike is only emitted if the membrane potential was below threshold at the beginning of a time step and is above at the end.

The general rule is this: If a model has a hard reset upon threshold crossing, then we need the min() to ensure that the potential does not exceed the threshold.

Reply:
Thanks for the clarification.